src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/Match.cs

```diff
@@ -162,7 +162,7 @@ private bool TryCreateReturnType(TexlNode regExNode, string regexPattern, IError
 
                 if (!startMatchHidden)
                 {
-                    propertyNames.Add(new TypedName(DType.Number, ColumnName_StartMatch));
+                    propertyNames.Add(new TypedName(DType.Decimal, ColumnName_StartMatch));
```

Comment:
>Decimal [](http://example.com/codeflow?start=58&length=7)

We should only change if Not(context.NumberIsFloat)

Reply:
Great call.  I backed out the regular expression changes for now.  The change in the compiler is trivial, but the change in the interpeter is more involved.  Regular expressions are not a part of Power Fx 1.0 so it can wait.  Tracked with https://github.com/microsoft/Power-Fx/issues/1816